test/ranking/ranking.test.ts

```diff
@@ -30,4 +31,64 @@ describe('ranking', () => {
       assert.deepEqual(group.items, []);
     });
   });
+
+  describe('comparatorFactory', () => {
+    it('should create a comparator that returns a score difference when passed an orderBy array', () => {
+      const specM1 = SpecQueryModel.build(
+        {
+          mark: Mark.POINT,
+          encodings: [
+            {channel: Channel.X, field: 'Q', type: Type.QUANTITATIVE},
+            {channel: Channel.Y, field: 'Q1', type: Type.QUANTITATIVE},
+          ]
+        },
+        schema,
+        DEFAULT_QUERY_CONFIG
+      );
+
+      const specM2 = SpecQueryModel.build(
+        {
+          mark: Mark.BAR,
+          encodings: [
+            {channel: Channel.X, field: 'N', type: Type.NOMINAL},
+            {channel: Channel.Y, field: 'Q', type: Type.QUANTITATIVE},
+          ]
+        },
+        schema,
+        DEFAULT_QUERY_CONFIG
+      );
+
+      const comparator = comparatorFactory(['aggregationQuality', 'effectiveness'], schema, DEFAULT_QUERY_CONFIG);
+      assert.isNumber(comparator(specM1, specM2));
```

Comment:
This doesn't tell if your comparator returns correct result!  

The test would pass even if you put a random number generator as the comparator function.